### ISSUE TITLE
C0325 (superfluous-parens) appears to not trigger when conents are a string

### ISSUE THREAD

**Issue by treyra** [pylint-dev/pylint#9878]
### Current problem

When superfluous-parens surround a string, no warning is raised. This can lead to errors in functions designed to test strings against tuples of acceptable values, such as when loading a bunch of configurations from a text file. 

```
def testMethod(testString):
    """
    Test method to demonstrate C0325 (superfluous-parens) gets triggered after if, but not in keywords
    """
    if (testString == "option1"): # superfluous-parens
        return 1
    if testString == ("option2"): # no warning
        return 2
    if testString in ("option1","option2"):
        return 3
    if testString in ("option4"): # no warning
        return 4
    if testString == "option": 
        raise ValueError("Did you forget to specify an option number?")
    return "None of the above"

print(testMethod("option"))
```

### Desired solution

Warning on ("string") after a keyword.

### Additional context

The example code (specifically the return 4 blocks) might arise when comparing against tuples of valid strings, and all but one option has been removed. If the comma is forgotten, then this method will return `4`, which might be unexpected to a user. This change would help avoid this issue in an evolving code base.

**Comment by kaidenwang0309-hub** [comment]
Hi, @Pierre-Sassoulas, I also tested this with integers and floats, and it seems that unless the parameters are surrounding a boolean, no warning is raised.  
```python
test_int = 1 
test_float = 0.1

if (test_int == 1): # raises warning
  print("hi")
if (test_int) == 1: # no warning
  print("hello")
if (test_float) == 0.1: # no warning
  print("goodbye")
if (test_int) is 1: # no warning
  print("hi")
if (test_float) is 0.1: # no warning
  print("hello")
```
This seems to imply that the implementation of C0325 only targets booleans, and refactoring the code to check all types would resolve the issue.

**Comment by ranadheerg** [comment]
I was checking an issue related to C0325 (superfluous-parens) 
https://github.com/pylint-dev/pylint/issues/10084 it seems related to this issue. I am looking at implementation of format.py to see if both can be fixed with a common PR.
